androidtv/basetv/basetv.py

```diff
@@ -201,8 +232,39 @@ def _cmd_hdmi_input(self):
         if self.DEVICE_ENUM == constants.DeviceEnum.ANDROIDTV and self.device_properties.get("sw_version", "") == "11":
             return constants.CMD_HDMI_INPUT11
 
+        # Is this an Android 12 device?
+        if self.DEVICE_ENUM == constants.DeviceEnum.ANDROIDTV and self.device_properties.get("sw_version", "") == "12":
+            return constants.CMD_HDMI_INPUT11
+
+        # Is this an Android 13 device?
+        if self.DEVICE_ENUM == constants.DeviceEnum.ANDROIDTV and self.device_properties.get("sw_version", "") == "13":
+            return constants.CMD_HDMI_INPUT11
+
         return constants.CMD_HDMI_INPUT
 
+    def _cmd_volume_set(self):
```

Comment:
I think it would be simpler to take `volume_level` as a parameter and return the formatted string.